gdal/frmts/fits/fitsdataset.cpp

```diff
@@ -599,6 +720,684 @@ GDALDataset *FITSDataset::Create( const char* pszFilename,
   }
 }
 
+/************************************************************************/
+/*                          WriteFITSInfo()                          */
+/************************************************************************/
+
+void FITSDataset::WriteFITSInfo()
+
+{
+  int status = 0;
+
+  const double PI = std::atan(1.0)*4;
+  const double DEG2RAD = PI / 180.;
+
+  double falseEast = 0;
+  double falseNorth = 0;
+
+  double cfactor, mres, mapres, UpperLeftCornerX, UpperLeftCornerY;
+  double crpix1, crpix2;
+
+/* -------------------------------------------------------------------- */
+/*      Write out projection definition.                                */
+/* -------------------------------------------------------------------- */
+    const bool bHasProjection =
+        pszProjection != nullptr && strlen(pszProjection) > 0;
+    if( bHasProjection )
+    {
+        OGRSpatialReference oSRS;
+        oSRS.importFromWkt( pszProjection );
+
+        // Set according to coordinate system (thanks to Trent Hare - USGS)
+
+        char object[16];
+        char ctype1[9], ctype2[9];
+        
+        const char* target = oSRS.GetAttrValue("DATUM",0);
```

Comment:
this could return a nullptr

Reply:
Check added in 41699ae24b983381a1a63715fbbae71c442059d8